src/Pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { ChangeEvent, useCallback, useContext, useEffect, useState } from 'react';
+import CardList from '../../components/CardList/CardList';
+import CreateCard from '../../components/CreateCard/CreateCard';
+import s from './MainPage.module.css';
+import { Link, useNavigate } from 'react-router-dom';
+import Pagination from '../../components/Pagination/Pagination';
+
+import ModalWindow from '../../components/Modal/Modal';
+
+import { useSelector } from 'react-redux';
+import { getPagesNum, getCurrentPage } from '../../Features/CardsSlice';
+
+export type RickAndMortyCardProps = {
+  id: number;
+  name: string;
+  gender: string;
+  species: string;
+  created: string;
+  status: string;
+  image: string;
+  episode?: [];
+  location?: { name: string; url: string };
+  origin?: { name: string; url?: string };
+  type?: string;
+  url?: string;
+};
+
```

Comment:
sepate types and interfaces, that's use in different modules in separate file types.ts